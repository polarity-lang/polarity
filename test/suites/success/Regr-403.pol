use "../../../std/data/eq.pol"
use "../../../std/data/bool.pol"
use "../../../std/codata/fun.pol"

#[transparent]
let foo : Fun(Bool, Bool)  {
    \ap(_,_,x) => x
}

let proof: Eq(a := Fun(Bool, Bool), foo, foo) {
    Refl(a:= Fun(Bool,Bool), foo)
}
