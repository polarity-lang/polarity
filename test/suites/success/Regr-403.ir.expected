use "../../../std/data/eq.pol"
use "../../../std/data/bool.pol"
use "../../../std/codata/fun.pol"

let foo { comatch { .ap(x1) => x1 } }

let proof { Refl(foo) }
